.github/workflows/matrix.yml

```
@@ -243,9 +243,30 @@ jobs:
         restore-keys: |
           ccache-${{ runner.os }}-${{ matrix.compiler }}--
     - uses: ammaraskar/gcc-problem-matcher@master
-    - name: build and test
+    - name: build
       if: ${{ env.SKIP == 'false' }}
-      run: bash ./build-scripts/build.sh
+      run: bash ./build-scripts/gha_compile_only.sh
+    - name: retrieve ccache debug files
+      run: |
+        find . -type f -name "*cache-*" -exec tar -rvf ccache.debug.tar {} \;
+    - name: compress ccache debug files
+      run: xz -v ccache.debug.tar
+    - name: upload ccache debug files
+      uses: actions/upload-artifact@v2
+      with:
+        name: ccache-debug-${{ runner.os }}-${{ matrix.compiler }}
+        path: ccache.debug.tar.xz
+        if-no-files-found: ignore
```

```suggestion
      run: bash ./build-scripts/gha_compile_only.sh
    - name: create ccache debug file archive
      run: |
        find . -type f -name "*cache-*" | tar -c --files-from - | zstd -of ccache.debug.tar.zst
    - name: upload ccache debug files
      uses: actions/upload-artifact@v2
      with:
        name: ccache-debug-${{ runner.os }}-${{ matrix.compiler }}
        path: ccache.debug.tar.zst
        if-no-files-found: ignore
```
These separate steps take half an hour by themselves, as long as running tests. This form should take less than a minute.